pulp_rpm/app/serializers.py

```diff
@@ -331,85 +343,205 @@ class Meta:
         help_text=_("Update name"),
-        allow_blank=True
+        allow_blank=True,
+        read_only=True
     )
     summary = serializers.CharField(
         help_text=_("Short summary"),
-        allow_blank=True
+        allow_blank=True,
+        read_only=True
     )
     version = serializers.CharField(
         help_text=_("Update version (probably always an integer number)"),
-        allow_blank=True
+        allow_blank=True,
+        read_only=True
     )
 
     type = serializers.CharField(
         help_text=_("Update type ('enhancement', 'bugfix', ...)"),
-        allow_blank=True
+        allow_blank=True,
+        read_only=True
     )
     severity = serializers.CharField(
         help_text=_("Severity"),
-        allow_blank=True
+        allow_blank=True,
+        read_only=True
     )
     solution = serializers.CharField(
         help_text=_("Solution"),
-        allow_blank=True
+        allow_blank=True,
+        read_only=True
     )
     release = serializers.CharField(
         help_text=_("Update release"),
-        allow_blank=True
+        allow_blank=True,
+        read_only=True
     )
     rights = serializers.CharField(
         help_text=_("Copyrights"),
-        allow_blank=True
+        allow_blank=True,
+        required=False
     )
     pushcount = serializers.CharField(
         help_text=_("Push count"),
-        allow_blank=True
+        allow_blank=True,
+        read_only=True
+    )
+    reboot_suggested = serializers.BooleanField(
+        help_text=_("Reboot suggested"),
+        required=False
     )
     pkglist = UpdateCollectionSerializer(
         source='collections', read_only=True,
-        many=True, help_text=_("List of packages")
+        many=True, help_text=_("List of packages"),
+        required=False
     )
     references = UpdateReferenceField(
         source='pk', read_only=True,
-        help_text=_("List of references")
+        help_text=_("List of references"),
+        required=False
     )
 
+    def create(self, validated_data):
+        """
+        Create UpdateRecord and its subclasses from JSON file.
+
+        Returns:
+            UpdateRecord instance
+
+        """
+        references = validated_data.pop("references", [])
+        pkglist = validated_data.pop("pkglist", [])
+        update_collection_packages_to_save = list()
+        update_references_to_save = list()
+        try:
+            update_record = super().create(validated_data)
+        except IntegrityError:
+            raise serializers.ValidationError("Advisory already exists in Pulp.")
+
+        for collection in pkglist:
+            packages = collection.pop("packages", [])
+            new_pkg = collection
+            new_pkg[PULP_UPDATE_COLLECTION_ATTRS.SHORTNAME] = new_pkg.pop(
+                "short", ""
+            )
+            coll = UpdateCollection(**new_pkg)
+            try:
+                coll.save()
+            except IntegrityError:
```

Comment:
why did you add the try/except here?

Reply:
to not save duplicates

Reply:
There is no uniqueness constraint on the UpdateCollection, IntegrityError will never happen in such cases.

Reply:
so it can be removed